pkg/apis/aws/cache/v1alpha1/replication_group_types_test.go

```diff
@@ -202,7 +202,7 @@ func TestNewReplicationGroupSpec(t *testing.T) {
 	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
 			got := NewReplicationGroupSpec(tc.properties)
-			if diff := deep.Equal(got, tc.want); diff != nil {
+			if diff := cmp.Diff(got, tc.want); diff != "" {
 				t.Errorf("got != want:\n%v", diff)
```

Comment:
I like this paradig, where all you need to print is `diff`!
Side note: typically  `t.Errorf()` should contain function under test name, i.e., `t.Errorf("NewReplicationGroupSpec() got != want:\n%v", diff)`

Reply:
Agreed on both counts - I think tests should always include the function they're calling, the `diff`, and not worry about also printing the `got` and `want` values specifically. I don't want to take on that refactoring in the scope of this PR though.